src/biotite/structure/io/pdbx/convert.py

```diff
@@ -650,31 +650,35 @@ def _find_matches(query_arrays, reference_arrays):
     `reference_arrays` where all query values match the reference counterpart.
     If no match is found for a query, the corresponding index is -1.
     """
-    match_masks_for_all_columns = np.stack(
-        [
-            query[:, np.newaxis] == reference[np.newaxis, :]
-            for query, reference in zip(query_arrays, reference_arrays)
-        ],
-        axis=-1,
-    )
-    match_masks = np.all(match_masks_for_all_columns, axis=-1)
-    query_matches, reference_matches = np.where(match_masks)
-
-    # Duplicate matches indicate that an atom from the query cannot
-    # be uniquely matched to an atom in the reference
-    unique_query_matches, counts = np.unique(query_matches, return_counts=True)
-    if np.any(counts > 1):
-        ambiguous_query = unique_query_matches[np.where(counts > 1)[0][0]]
-        raise InvalidFileError(
-            f"The covalent bond in the 'struct_conn' category at index "
-            f"{ambiguous_query} cannot be unambiguously assigned to atoms in "
-            f"the 'atom_site' category"
-        )
+    # Convert reference arrays to a dictionary for O(1) lookups
+    reference_dict = {}
+    unambiguously_keys = set()
+    for idx, col in enumerate(np.stack(reference_arrays, axis=-1)):
+        ref_key = "_".join(map(str, col))
```

Comment:
Is there a reason to convert specifically to string here? I assume we could save computation time if we used just a tuple of the original values.